aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp

```diff
@@ -86,12 +86,22 @@ ideep::tensor itensor_view_from_dense(const Tensor& tensor) {
   TORCH_CHECK(
       tensor.layout() == Layout::Strided,
       "itensor_view_from_dense expects dense tensor input");
-  TORCH_CHECK(tensor.scalar_type() == ScalarType::Float,
+  TORCH_CHECK((tensor.scalar_type() == ScalarType::Float) ||
+              (tensor.scalar_type() == ScalarType::BFloat16),
```

Comment:
It would be better to remove this TORCH_CHECK, test if it's bfloat16 before going in the bfloat16 branch, and error in the else branch. Saves you a condition branch

Reply:
Mkldnn code does not distinguish data type like bfloat16 or float from API perspective. We won't split mkldnn_gelu_float or mkldnn_gelu_bfloat16 in mkldnn branch. It is more convenient to check outside API.

Reply:
I mean something different.

```
  TORCH_INTERNAL_ASSERT(at::impl::variable_excluded_from_dispatch());
  if (tensor.scalar_type() == ScalarType::Float){
    return {{tensor.sizes().vec(),
            ideep::tensor::data_type::f32,
            tensor.strides().vec()},
            tensor.template data_ptr<float>()};
  }
  else if (tensor.scalar_type() == ScalarType::BFloat16) {
    return {{tensor.sizes().vec(),
            ideep::tensor::data_type::bf16,
            tensor.strides().vec()},
            tensor.template data_ptr<BFloat16>()};
  }
else {
  TORCH_CHECK(false, "itensor_view_from_dense expects float/bfloat16 tensor input");
}
```

Reply:
Got it. Done.